File: src/SkyBlock/enchants/block/Booster.php
<?php


namespace SkyBlock\enchants\block;


use pocketmine\block\BlockToolType;
use pocketmine\entity\effect\EffectInstance;
use pocketmine\entity\effect\VanillaEffects;
use pocketmine\event\block\BlockBreakEvent;
use pocketmine\item\Pickaxe;
use pocketmine\player\Player;

class Booster extends BaseBlockBreakEnchant {

    static int $id = 139;

    public function isApplicableTo(Player $holder, int $level = 0) : bool {
        if ($holder->getInventory()->getItemInHand() instanceof Pickaxe or $holder->getInventory()->getItemInHand() instanceof \SkyBlock\item\Pickaxe) {
            return mt_rand(1, 20) <= $level;
        }
        return false;
    }

    public function getLevel(int $level = 1) : int {
        return (($int = ceil($level / 2)) > 7) ? 7 : $int;
    }


    public function onActivation(Player $player, BlockBreakEvent $ev, int $enchantmentlevel) : void {
        if ($ev->getBlock()->getBreakInfo()->getToolType() === BlockToolType::PICKAXE) {
            $effect = new EffectInstance(VanillaEffects::SPEED(), $enchantmentlevel * 2 * 20, $this->getLevel($enchantmentlevel), false);
            $player->getEffects()->add($effect);
        }
    }
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/block/Booster.java
package dev.skyblock.enchants.block;

import dev.skyblock.enchants.BaseBlockEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.block.BlockBreakEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import java.util.concurrent.ThreadLocalRandom;

public class Booster extends BaseBlockEnchant {
    public static final int ID = 139;

    public Booster() {
        super(ID, "Booster", 5);
    }

    @Override
    public boolean isApplicableTo(Player player, int level) {
        return ThreadLocalRandom.current().nextInt(1, 26) == 1;
    }

    @Override
    public void onActivation(Player player, BlockBreakEvent ev, int level) {

    }
}
